Upgrade to Kodi 18.6 and support Yocto release Dunfell and Zeus

leon-anavi opened the pull request:
Hi,

This GitHub pull request includes the following changes:

* Add support for Yocto releases Zeus and Dunfell, including requirement for meta-python2 which provides `classes/python-dir.bbclass`
* Apply libffi bbappend to newer versions
* Upgrade Kodi to to the latest stable relase "Leia" 18.6

Tested on Raspberry Pi 3 and verified it builds for QEMUx86-64.

Best regards,
Leon

leon-anavi committed:
libffi_3.2.1.bbappend: remove

Remove the bbappend for libffi because this fix for the header is
no longer needed for building kodi with newer versions of libffi.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

leon-anavi committed:
conf/layer.conf: Compatibility for zeus & dunfell

Add compatibility for Yocto releases zeus add dunfell.

Add Yocto/OE layer meta-python2 as a dependency because it
provides classes/python-dir.bbclass.

Remove compatibility for older Yocto releases because they
provide an older version of libffi which requires a bbappend.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

leon-anavi committed:
kodi: upgrade to 18.6

Upgrade Kodi to the latest stable relase "Leia" 18.6.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

koenkooi commented:
We had some earlier pull requests for libffi stating the bbappend wasn't needed anymore for newer versions.
Did you run into issues where the main recipe didn't install the header?

leon-anavi commented:
Hi @koenkooi,

Yes, we can build Kodi without the bbappend file for libffi. I've update my GitHub pull request and removed it. However there is caveat, I've also removed the support for sumo, thud and warrior because they provide the older version of libffi.

Best regards,
Leon

dev-0x7C6 commented:
Hi @leon-anavi, 

I have similar fixes in (https://github.com/dev-0x7C6/meta-kodi/commits/master-next). I'm queuing all my work to meta-kodi:devel.

My recommendation is to target devel branch instead of master.

Cheers,
Bartłomiej

leon-anavi commented:
Hi @dev-0x7C6,

I've noticed the devel branch after I made this GitHub pull request. The default branch points to sumo and intuitively I switched to master. As I don't have previous experience with meta-kodi, I am kind of curious why doesn't meta-kodi follow the pretty much standard branch naming conventions for branches (I mean like meta-openembedded for example with branch for each release and master/master-next)?

Thanks,
Leon

dev-0x7C6 commented:
@leon-anavi 

> I've noticed the devel branch after I made this GitHub pull request. The default branch points to sumo and intuitively I switched to master.

Sorry to hear that :-(

> As I don't have previous experience with meta-kodi, I am kind of curious why doesn't meta-kodi follow the pretty much standard branch naming conventions for branches (I mean like meta-openembedded for example with branch for each release and master/master-next)

Following this workflow need constant work with OE master-next branches and testing/patching against them. This is time consuming and it's a better approach for layers with more active developers.

There is a story about devel branch #14
